applications/MeshingApplication/CMakeLists.txt

```diff
@@ -56,79 +40,35 @@ if(${KRATOS_BUILD_TESTING} MATCHES ON)
-            SET(MMG2D_INCLUDE_DIR ${MMG_ROOT}/include/mmg/mmg2d/)
-        ENDIF(NOT DEFINED MMG2D_INCLUDE_DIR)
-        IF(NOT DEFINED MMG3D_INCLUDE_DIR)
-            SET(MMG3D_INCLUDE_DIR ${MMG_ROOT}/include/mmg/mmg3d/)
-        ENDIF(NOT DEFINED MMG3D_INCLUDE_DIR)
-        IF(NOT DEFINED MMGS_INCLUDE_DIR)
-            SET(MMGS_INCLUDE_DIR ${MMG_ROOT}/include/mmg/mmgs/)
-        ENDIF(NOT DEFINED MMGS_INCLUDE_DIR)
-        IF(NOT DEFINED MMG_LIBRARY)
-            SET(MMG_LIBRARY ${MMG_ROOT}/lib/${MMG_LIBRARY_PREFIX}mmg.${MMG_LIBRARY_EXTENSION})
-        ENDIF(NOT DEFINED MMG_LIBRARY)
-        IF(NOT DEFINED MMG2D_LIBRARY)
-            SET(MMG2D_LIBRARY ${MMG_ROOT}/lib/${MMG_LIBRARY_PREFIX}mmg2d.${MMG_LIBRARY_EXTENSION})
-        ENDIF(NOT DEFINED MMG2D_LIBRARY)
-        IF(NOT DEFINED MMG3D_LIBRARY)
-            SET(MMG3D_LIBRARY ${MMG_ROOT}/lib/${MMG_LIBRARY_PREFIX}mmg3d.${MMG_LIBRARY_EXTENSION})
-        ENDIF(NOT DEFINED MMG3D_LIBRARY)
-        IF(NOT DEFINED MMGS_LIBRARY)
-            SET(MMGS_LIBRARY ${MMG_ROOT}/lib/${MMG_LIBRARY_PREFIX}mmgs.${MMG_LIBRARY_EXTENSION})
-        ENDIF(NOT DEFINED MMGS_LIBRARY)
-    ENDIF (MMG_ROOT)
+        set(MMG_BUILDDIR ${MMG_ROOT})
+        set(MMG_LIBDIR ${MMG_ROOT}/lib)
+    ENDIF(MMG_ROOT)
+
+    find_package(MMG)
+    find_package(MMG2D)
+    find_package(MMG3D)
+    find_package(MMGS)
+
+    include_directories(${MMG_INCLUDE_DIRS} ${MMG2D_INCLUDE_DIRS} ${MMG3D_INCLUDE_DIRS} ${MMGS_INCLUDE_DIRS})
+    set(KRATOS_MESHING_APPLICATION_EXTRA_LIBS ${KRATOS_MESHING_APPLICATION_EXTRA_LIBS} ${MMG_LIBRARIES} ${MMG2D_LIBRARIES} ${MMG3D_LIBRARIES} ${MMGS_LIBRARIES})
+
+    MESSAGE(STATUS "Compilation with MMG: ${MMG_LIBRARIES} ${MMG_INCLUDE_DIRS}")
+    MESSAGE(STATUS "Compilation with MMG2D: ${MMG2D_LIBRARIES} ${MMG_INCLUDE_DIRS}")
+    MESSAGE(STATUS "Compilation with MMG3D: ${MMG3D_LIBRARIES} ${MMG_INCLUDE_DIRS}")
+    MESSAGE(STATUS "Compilation with MMGS: ${MMGS_LIBRARIES} ${MMG_INCLUDE_DIRS}")
 
     # Add definitions
     add_definitions( -DINCLUDE_MMG )
-
-    INCLUDE_DIRECTORIES(${MMG_INCLUDE_DIR})
-    INCLUDE_DIRECTORIES(${MMG2D_INCLUDE_DIR})
-    INCLUDE_DIRECTORIES(${MMG3D_INCLUDE_DIR})
-    INCLUDE_DIRECTORIES(${MMGS_INCLUDE_DIR})
-
-    target_link_libraries(KratosMeshingCore PUBLIC KratosCore tetMeshOpt ${MMG_LIBRARY} ${MMG2D_LIBRARY} ${MMG3D_LIBRARY} ${MMGS_LIBRARY})
-    
-    message("Installed MMG = " ${MMG_LIBRARY})
-    install(FILES ${MMG_LIBRARY} DESTINATION libs)
-    message("Installed MMG2D = " ${MMG2D_LIBRARY})
-    install(FILES ${MMG2D_LIBRARY} DESTINATION libs)
-    message("Installed MMG3D = " ${MMG3D_LIBRARY})
-    install(FILES ${MMG3D_LIBRARY} DESTINATION libs)
-    message("Installed MMGS = " ${MMGS_LIBRARY})
-    install(FILES ${MMGS_LIBRARY} DESTINATION libs)
 else(${INCLUDE_MMG} MATCHES ON)
-    target_link_libraries(KratosMeshingCore PUBLIC KratosCore tetMeshOpt)
+    list(FILTER KRATOS_MESHING_APPLICATION_CORE EXCLUDE REGEX "mmg")
 endif(${INCLUDE_MMG} MATCHES ON)
```

Comment:
Since now the GLOB_RECURSE is used to find the cpp files, here any file with "mmg" is removed from the list if INCLUDE_MMG=OFF. Do you know any better way to this? (apart from selecting manually the cpp files as it was before)

Reply:
I do, will post shortly

Reply:
Ok done! There are many file changes now but most of them are the renaming and update of includes.